core/src/main/java/com/graphhopper/storage/CHGraphImpl.java

```diff
@@ -176,20 +155,23 @@ int weightToWeightFlags(long edgePointer, double weight) {
                     " nodeB " + nodeAccess.getLat(getNodeB(edgePointer)) + "," + nodeAccess.getLon(getNodeB(edgePointer)));
             weight = MIN_WEIGHT;
         }
-        if (weight > MAX_WEIGHT)
-            weightInt = MAX_WEIGHT_31;
+        if (weight >= WEIGHT_INFINITY)
+            weightInt = (int) WEIGHT_LONG_INFINITY; // negative
```

Comment:
And actually it would be interesting to know if there are many of these shortcuts. We might as well not add these shortcuts at all since they use infinite weight anyway (so they will never be part of a route). Maybe we can just log how many of these cases we found?

Reply:
I would make `MAX_WEIGHT` public and use it in `Node/EdgeBasedShortcutInserter` to count how many of these shortcuts there are and log this number for now. This way we can figure out how frequently this happens.

Reply:
(I can also do this after we merge this PR of course)

Reply:
> I did a brute force test setting/getting shortcut weights from 1 to (long) Integer.MAX_VALUE << 1 and it worked, but I wonder if this could be JVM implementation specific or something?

Not sure what do you mean here exactly. Do you mean the cast into long or the cast of WEIGHT_LONG_INFINITY into int or the shift? All those operations are well defined and do not depend on the JVM - we rely on those low level stuff elsewhere like in landmark storage etc. (but I always need some time to sink again deep into this topic like what was the unsigned right shift and what was signed etc :) )

> (I can also do this after we merge this PR of course)

Yes, probably one log entry would be best where we loop through all shortcuts

Reply:
> All those operations are well defined and do not depend on the JVM

Ok, this is what I was wondering.